tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/DataType.java

```diff
@@ -67,50 +53,39 @@ public int byteSize() {
     return byteSize;
   }
 
-  /** Corresponding value of the TF_DataType enum in the TensorFlow C API. */
-  int c() {
-    return value;
+  /**
+   * Returns a readable name for this type
+   */
+  public String name() {
+    return name;
   }
 
-  // Cached to avoid copying it
-  private static final DataType[] values = values();
-
-  static DataType fromC(int c) {
-    for (DataType t : values) {
-      if (t.value == c) {
-        return t;
-      }
-    }
-    throw new IllegalArgumentException(
-        "DataType " + c + " is not recognized in Java (version " + TensorFlow.version() + ")");
+  /**
+   * Returns the numeric code for this datatype, as recognized by the native library (C API)
+   */
+  int nativeCode() {
+    return nativeCode;
   }
 
   /**
-   * Returns the DataType of a Tensor whose elements have the type specified by class {@code c}.
+   * Maps a tensor to a data structure for manipulating elements of this type.
    *
-   * @param c The class describing the TensorFlow type of interest.
-   * @return The {@code DataType} enum corresponding to {@code c}.
-   * @throws IllegalArgumentException if objects of {@code c} do not correspond to a TensorFlow
-   *     datatype.
+   * @param tensor tensor to map
+   * @return data structure of elements of this type
    */
-  public static DataType fromClass(Class<?> c) {
-    DataType dtype = typeCodes.get(c);
-    if (dtype == null) {
-      throw new IllegalArgumentException(
-          c.getName() + " objects cannot be used as elements in a TensorFlow Tensor");
-    }
-    return dtype;
+  T map(Tensor tensor) {
```

Comment:
This takes a bare `Tensor` type, can it have a stronger bound? Either <? extends TType>  or even `<T extends TType>`.

Reply:
why I get no warning about this stuff... :-/ I'll double-check everywhere, thanks)

Reply:
We could turn on compiling with warnings, then maven would spray them everywhere.

Reply:
You just made me realize that there were a ton of lint checks applied in the previous Bazel build that we forgot to migrate to Maven, we need to restore them... I'll create a separate issue about this.

Reply:
Here: https://github.com/tensorflow/java/issues/7